earn/src/data/BorrowerNft.ts

```diff
@@ -110,53 +116,49 @@ export async function fetchListOfBorrowerNfts(
       const isInCorrectPool = filterParams?.validUniswapPool === undefined || borrowersInCorrectPool.has(borrower);
       return (areManagersValid || (filterParams?.includeFreshBorrowers && !isInUse)) && isInCorrectPool;
     })
-    .map(([borrower, managerSet]) => borrower);
+    .map(([borrower, _managerSet]) => borrower);
 
-  // Fetch decoded SSTORE2 data from the BorrowerNFT (tokenIds in a specific order)
   const orderedTokenIds = (await borrowerNftContract.tokensOf(userAddress)) as BigNumber[];
   const orderedTokenIdStrs = orderedTokenIds.map((id) => '0x' + id.toHexString().slice(2).padStart(44, '0'));
 
-  const tokenIds = borrowers.map((borrower) => orderedTokenIdStrs.find((x) => x.startsWith(borrower.toLowerCase()))!);
-  const indices = borrowers.map((borrower) =>
-    orderedTokenIdStrs.findIndex((x) => x.startsWith(borrower.toLowerCase()))
-  );
-
-  return { borrowers, tokenIds, indices };
+  return borrowers.map((borrower) => {
+    return {
+      borrowerAddress: borrower,
+      tokenId: orderedTokenIdStrs.find((x) => x.startsWith(borrower.toLowerCase()))!,
+      index: orderedTokenIdStrs.findIndex((x) => x.startsWith(borrower.toLowerCase())),
```

Comment:
We could throw an error if either of these fails to find a matching entry, rather than returning undefined / -1